docs/blog/2024-05-01-active-learning.md

```diff
@@ -0,0 +1,75 @@
+tags: [active learning, uncertainty sampling, deep dive]
+image: ./images/active-learning/social.png
+hide_table_of_contents: false
+---
+
+At Groundlight, we train each detector's machine learning (ML) model on images that have been manually labeled with correct responses. However, these labels are expensive to collect because they require human review. Given that detectors are frequently applied to streams of images that change slowly over time, reviewing all images as they arrive is likely to result in wasted effort labeling images that are similar and add little new information to the training set.
+
+To avoid unnecessary labeling and save customers money, Groundlight uses **[active learning](https://en.wikipedia.org/wiki/Active_learning_(machine_learning))**, a machine learning protocol in which an ML model plays an active role in determining which images get manually labeled for training. With active learning, only informative images are prioritized for review, making it possible to label small a subset of the available data but train a model that's roughly as good as one trained with all the data labeled [\[Settles, 2009\]](https://minds.wisconsin.edu/handle/1793/60660).
+
+<!-- truncate -->
+
+The variant of active learning we use Groundlight is based on **[uncertainty sampling](https://lilianweng.github.io/posts/2022-02-20-active-learning/#uncertainty-sampling)**, a well studied and effective method that can be used in either the streaming setting or the pool-based setting in which there exists a large reservoir of unlabeled examples to draw from. We operate in the stream-based setting, where images arrive one at a time and we have to decide in the moment whether to escalate an image for review.
+
+Uncertainty sampling works as follows. Imagine we have a detector that processes a stream of images arriving one by one. The detector's ML model is trained on all previously labeled images from the stream. When a new image arrives, the model makes its best guess prediction for the image and also reports its confidence in its prediction. The confidence is expressed as a probability (a number between zero and one) that its prediction is correct. In uncertainty sampling, we escalate and label those images whose predictions have low confidence, and largely leave images with confident predictions unlabeled. In this way, we avoid most of the expense and effort of labeling images whose predictions are likely correct, but still continue to improve the model on images it's unsure of.
+
+As an example, the images shown below were sent to a detector that identifies the presence of dogs in and around an indoor swimming pool. The model reports with 95% confidence that there is a dog in the image on the left. But it is less confident in its response for the image on the right, saying there is no dog present with only 75% confidence. There is in fact a dog in that image, at the back left corner of the swimming pool, but it is difficult to see.
```

Comment:
Should we hyperlink to our camera feed provider?  Not necessary I think, but maybe polite.

Reply:
yes, good idea. will do